### PR TITLE
Created a functioning Docker container

### PR DESCRIPTION
**Pull request by mhanberry1**
The Dockerfile in `ubuntu-docker/1204/` can be used to build a functional docker container.

The key was to create a non-root user inside of the container and compile the code in that user's home directory. In addition, that user should be used to execute nsim.

I'm not 100% sure why this is necessary, but this approach fixes the issues that were observed in https://github.com/fangohr/nmag-containers/issues/1

### CODE REVIEW

File: ubuntu-docker/1204/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+You can execute a command from the container using `docker run nmag <command>`.
+For example:
+```
+docker run nmag nsim
+```
+To execute a command on a file in the current directory or a child of the current directory, use the format `docker run -v $PWD:. nmag <command> <dilename>`.
+For example:
+```
+docker run -v $PWD:. nmag nsim simulation.py
+```
```
Comment (fangohr) [inline ubuntu-docker/1204/README.md:24]: This command doesn't work for me.

<pre>
base) > $ docker run -v `pwd`:. nmag nsim simulation.py                                                                [±master ✓]
docker: Error response from daemon: invalid volume specification: '/Users/fangohr/git/nmag-containers/ubuntu-docker/1204:.': invalid mount config for type "bind": invalid mount path: '.' mount path must be absolute.
See 'docker run --help'.
</pre>

Should we have a dedicated directory (`/io`) inside the container, to which we mount the local path?

**Comment by fangohr** [comment]
Beautiful! Overall, this works great - thanks for figuring out the problem!

I have only two additions:

(i) I made a comment regarding the mount path

(ii) Is it worth having a simple nmag script (`simple.py`) in the same folder as the Dockerfile, just so that people can test this conveniently?

[Long term, we need to update the wget address for the nmag-0.2.1.tar.gz file.]

**Comment by mhanberry1** [comment]
@fangohr Please review the following, and merge my code if everything looks good.

> (i) I made a comment regarding the mount path

Addressed by 7b0d2a452cd64152370ad135287e1188e6b51f1c

> (ii) Is it worth having a simple nmag script (`simple.py`) in the same folder as the Dockerfile, just so that people can test this conveniently?

Addressed by de7b9494f6a5ded559c8f272fe6b2d7171447b80 and cdbdf32fd10cf20bce3229962b087914dafb6e2e

> [Long term, we need to update the wget address for the nmag-0.2.1.tar.gz file.]

Addressed by e89cb56402e5c662fedf57ebdf396d326b869588

I've also updated the main README.md file to reflect the full offering of functional containers. (7fb0ce025f7c57ea81d453f44f06e0b42b74fb33 and e509dbfc8b12b38002e0ee2233760d9509c5e25b)

File: ubuntu-docker/1204/README.md (hunk 1 above)
Comment (mhanberry1) [reply ubuntu-docker/1204/README.md:24]: Resolved by 7b0d2a452cd64152370ad135287e1188e6b51f1c.

**Comment by fangohr** [comment]
Excellent - thank you!

I like all the changes. 

I pointed out one missing -v in the README.

I also like the new `using` command examples to get into the interactive container, with and without mounting the current directory.

I have one more suggestion: Inside the container, can we activate `bash` by default? I don't know what the current default shell is, but it doesn't provide autocompletion (of `nsim`, and `sphere1.py`) etc.
(I hope I have explained this sufficiently clearly.)

**Comment by mhanberry1** [comment]
@fangohr

> I pointed out one missing -v in the README.

Addressed by 72e8ce173b185a867fa918a6a06f2b6cdebea161

> I have one more suggestion: Inside the container, can we activate `bash` by default? I don't know what the current default shell is, but it doesn't provide autocompletion (of `nsim`, and `sphere1.py`) etc.

Addressed by 9887a50922b7012392f0c57f9e03bf950875ff0a

**Comment by fangohr** [comment]
Perfect!

**Pull request merged**